src/constants/messageTypes.ts

```diff
@@ -0,0 +1,6 @@
+import type { CustomMessageType } from '../types/types';
+
+export const CUSTOM_MESSAGE_TYPE: Record<'date' | 'intro', CustomMessageType> = {
+  date: 'message.date',
+  intro: 'channel.intro',
+};
```

Comment:
```suggestion
export const CUSTOM_MESSAGE_TYPE = {
  date: 'message.date',
  intro: 'channel.intro',
} as const;
```

Reply:
That `Record` type casting would mean that `CUSTOM_MESSAGE_TYPE.date` could be either `message.date` or `channel.intro`:
![image](https://user-images.githubusercontent.com/43254280/163393157-ad3343b6-b374-4f0d-9af9-98a85c6abd02.png)
Let's make it type constant so the types look like this:
![image](https://user-images.githubusercontent.com/43254280/163393357-7db6a9b6-70b5-488d-8ca6-aa69cea5f62d.png)

Reply:
Added const assertion